frontend/src/components/details-page/collapse-block/collapse-block.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC, ReactNode, useEffect, useState } from 'react';
+
+import { Collapse } from '@mui/material';
+
+import styles from './styles.module.scss';
+
+interface CollapseBlockProps {
+  caption: string;
+  children: ReactNode;
+  open: boolean;
+}
+
+export const CollapseBlock: FC<CollapseBlockProps> = (props) => {
+  const { caption, children, open } = props;
+  const [collapseOpen, setCollapseOpen] = useState(open);
+
+  useEffect(() => {
+    setCollapseOpen(open);
+  }, [open]);
+
+  const toggleCollapse = (): void => {
+    setCollapseOpen(!collapseOpen);
+  };
```

Comment:
How should the parent know if the CollapseBlock was closed? Because I clearly see that the local `collapseOpen` can out-of-sync with the `props.open`

Reply:
The 'open' prop is needed in two cases:
* setting the initial state
* forced opening of the collapse when user choosing a configuration
* in other cases, the user can independently regulate the collapse without informing the parent